### PR TITLE
Place a bid in an auction

### LINKED ISSUE #1478: Auctions - Place Bid
#### What would you like to be added:
image

### PR DESCRIPTION
**Pull request by pierregee**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- This PR will let a user place a bid

**Out of Scope:**
- Quick bid
- Full Details bottom sheet in PlaceBidScreen

#### Which issue(s) does this PR fixes?:

Fixes #1478

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+          minNextBid={minNextBidInToken}
+          totalAuctionValue={totalLoanAmountInUSD}
+        />
+
+        <WalletTextInput
+          autoCapitalize='none'
+          onChange={(e: NativeSyntheticEvent<TextInputChangeEventData>) => {
+            setBidAmount(e.nativeEvent.text)
+          }}
+          placeholder='0.00'
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx:85]: Recommend to use `Enter an amount` to be consistent with all number input

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+          inputType='numeric'
+          valid={isValidMinBid}
+          inlineText={{
+            type: 'error',
+            text: isValidMinBid ? undefined : translate('screens/PlaceBidScreen', 'Bid amount is lower than required')
+          }}
+        >
+          <SetAmountButton
+            amount={new BigNumber(minNextBidInToken)}
+            onPress={onBidMinAmount}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx:100]: I think pressing `MIN. BID` button should always input the `min. next bid` even though user doesnt have enough balance. And then we'll display the corresponding validation. What do you think?

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/ConfirmPlaceBidScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    if (hasPendingBroadcastJob && currentBroadcastJob !== undefined && currentBroadcastJob.submitButtonLabel !== undefined) {
+      return currentBroadcastJob.submitButtonLabel
+    }
+    return 'PLACING BID'
+  }
+
+  return (
+    <ThemedScrollView
+      testID='confirm_place_bid_screen'
+      contentContainerStyle={tailwind('py-6 pb-8 px-4')}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/screens/ConfirmPlaceBidScreen.tsx:94]: Let's remove the unnecessary padding at the side and top
image

File: mobile-app/app/screens/AppNavigator/screens/Auctions/AuctionNavigator.tsx [hunk 4]
```diff
@@ -66,6 +80,32 @@ export function AuctionsNavigator (): JSX.Element {
...
         }}
       />
+
+      <AuctionsStack.Screen
+        component={PlaceBidScreen}
+        name='PlaceBidScreen'
+        options={{
+          headerTitle: () => (
+            <HeaderTitle
+              text={translate('screens/AuctionScreen', 'Place bid') + ' (Beta)'} // TODO: remove beta from title
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/AuctionNavigator.tsx:90]:
```suggestion
              text={translate('screens/AuctionScreen', 'Place Bid') + ' (Beta)'} // TODO: remove beta from title
```
for screen title we are using title case

File: mobile-app/app/screens/AppNavigator/screens/Auctions/AuctionNavigator.tsx [hunk 5]
```diff
@@ -66,6 +80,32 @@ export function AuctionsNavigator (): JSX.Element {
...
+        }}
+      />
+
+      <AuctionsStack.Screen
+        component={ConfirmPlaceBidScreen}
+        name='ConfirmPlaceBidScreen'
+        options={{
+          headerTitle: () => (
+            <HeaderTitle
+              text={translate('screens/AuctionScreen', 'Confirm place bid') + ' (Beta)'} // TODO: remove beta from title
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/AuctionNavigator.tsx:103]:
```suggestion
              text={translate('screens/AuctionScreen', 'Confirm Place Bid') + ' (Beta)'} // TODO: remove beta from title
```

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/ConfirmPlaceBidScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+        testID='title_auction_detail'
+        text={translate('screens/ConfirmPlaceBidScreen', 'AUCTION DETAILS')}
+      />
+      <NumberRow
+        lhs={translate('screens/ConfirmPlaceBidScreen', 'Total auction value (USD)')}
+        rhs={{
+          testID: 'total_auction_value',
+          value: totalAuctionValue,
+          suffixType: 'text',
+          suffix: '$'
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/screens/ConfirmPlaceBidScreen.tsx:134]:
```suggestion
          prefix: '$'
```

File: mobile-app/app/screens/AppNavigator/screens/Auctions/components/EmptyAuctionsScreen.tes.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { render } from '@testing-library/react-native'
```
[the remark this answers is not in the document]
Comment (pierregee) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/components/EmptyAuctionsScreen.tes.tsx]: This was a merge issue. It's a duplicate of EmptyAuction.tsx. Deleted as it is unused

**Review by kyleleow — CHANGES_REQUESTED** [review]

**Comment by kyleleow** [comment]
Since auction detail screen is ready, we can connect place bid function from there?

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    navigation.navigate('ConfirmPlaceBidScreen', {
+      batch,
+      bidAmount: new BigNumber(bidAmount),
+      estimatedFees,
+      totalAuctionValue: totalLoanAmountInUSD,
+      vault
+    })
+  }
+
+  const isValidMinBid = new BigNumber(bidAmount).gte(minNextBidInToken)
```
[the remark this answers is not in the document]
Comment (pierregee) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx:63]: Done in quick bid pr

File: mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx (hunk 2 above)
Comment (pierregee) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/screens/PlaceBidScreen.tsx:100]: Come to think of it, it says `MIN. BID`, I agree. Will update

**Pull request closed without merging**